src/store/ChartStore.js

```diff
@@ -256,36 +256,41 @@ class ChartStore {
 
         window.addEventListener('resize', this.resizeScreen.bind(this));
 
-        stxx.append('updateChartData', this.updateComparisons);
+        stxx.append('createDataSet', this.updateComparisons.bind(this));
     }
 
-    updateComparisons = (...args) => {
-        /* createDataSet/updateChartData sends more than ten updates per tick.
-            This is to avoid that.
-            Happens only for line chart because of animation
-        */
-        if (args[2] && !args[2].firstLoop) {return;}
-
+    @action.bound updateComparisons(...args) {
         let stx = this.context.stx;
-        let q = stx.currentQuote();
-        const comparisons = [];
-        if (q) {
-            for (const s in stx.chart.series) {
-                const d = stx.chart.series[s];
-                const p = d.parameters;
-                const price = d.lastQuote ? d.lastQuote.Close : undefined;
-
-                comparisons.push({
-                    color: p.color,
-                    price,
-                    symbolObject: p.symbolObject,
-                });
+        const comparisonSymbolsKeys = Object.keys(stx.chart.series);
+        if (comparisonSymbolsKeys.length !== this.comparisonSymbols.length) {
+            const comparisons = [];
+            let q = stx.currentQuote();
+            if (q) {
+                for (const s of comparisonSymbolsKeys) {
+                    const d = stx.chart.series[s];
+                    const p = d.parameters;
```

Comment:
@brucebinary please avoid single letter variables ... instead of `p` use `prm` or param`. instead of `s` use `sybl` or `key` ... instead of `d` use `seri`